src/Parser/Parser.hs

```diff
@@ -36,14 +36,14 @@ import Data.Functor.Identity(Identity)
 data ParState =
    PS {
        ctype      :: Maybe Type,       -- ^ type of current object being parsed
-       whileNames :: (Name, [Name]),   -- ^ arguments to the last-parsed function, needed for While
```

Comment:
I think the first `Name` was there to store the wrapping function's name, however, that was not used for anything. Since there is not function name in the case of `let`, I removed it altogether.